flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -123,13 +123,22 @@ private void openCli(String sessionId, Executor executor) {
                             CliOptionsParser.OPTION_FILE.getOpt()));
         }
 
-        boolean isInteractiveMode = !hasSqlFile && !hasUpdateStatement;
-
         try (CliClient cli = new CliClient(sessionId, executor, historyFilePath)) {
-            if (isInteractiveMode) {
-                cli.open();
+            if (options.getInitFile() != null) {
+                boolean success = cli.executeInitialization(readFromURL(options.getInitFile()));
+                if (!success) {
+                    System.out.println(
+                            "Fail to execute init file... Please refer to the LOG for detailed error messages.");
+                    return;
+                } else {
+                    System.out.println("Succeed to execute init file. Enter execution phase...");
```

Comment:
rephrase to "Successfully initialized from sql script: " + options.getInitFile()